app/xds_api/urls.py

```diff
@@ -7,15 +7,21 @@
 
 app_name = 'xds_api'
 urlpatterns = [
-    path('configuration/', views.XDSConfigurationView.as_view(),
-         name='xds-configuration'),
-    path('ui-configuration/', views.XDSUIConfigurationView.as_view(),
-         name='xds-ui-configuration'),
-    path('spotlight-courses', views.get_spotlight_courses,
-         name='spotlight-courses'),
-    path('auth', include('knox.urls')),
-    path('auth/register', views.RegisterView.as_view(), name='register'),
-    path('auth/login', views.LoginView.as_view(), name='login'),
-    path('courses/<str:course_id>/', views.get_courses,
-         name='get_courses'),
+     path('configuration/', views.XDSConfigurationView.as_view(),
+          name='xds-configuration'),
+     path('ui-configuration/', views.XDSUIConfigurationView.as_view(),
+          name='xds-ui-configuration'),
+     path('spotlight-courses', views.get_spotlight_courses,
+          name='spotlight-courses'),
+     path('auth', include('knox.urls')),
+     path('auth/register', views.RegisterView.as_view(), name='register'),
+     path('auth/login', views.LoginView.as_view(), name='login'),
+     path('courses/<str:course_id>/', views.get_courses,
+          name='get_courses'),
+     path('interest-lists/', views.interest_lists,
+          name='interest-lists'),
+     path('interest-lists/<int:list_id>', views.single_interest_list,
+          name='single-interest-list'),
+     path('add-course-to-lists/', views.add_course_to_lists,
```

Comment:
This is an odd URL compared to the related ones -- we seem to have a combination of noun-oriented URLs and verb-oriented URLs -- we should pick one style and try to stick with it.  Can this be a POST to the `interest-lists` endpoint, or a POST to `interest-lists/{:id}/courses`?

Reply:
I was conflicted with this one because the call itself is an odd call. We already have a post for interest-lists which creates a new list. This call is to add a single course to multiple lists on the front end (checkboxes). It didn't really fit into the typical API crud operations. Maybe we this should be "/api/experience/{:id}" ?

Reply:
When designing REST endpoints I've often created "two way streets" to support relationships between resources.  It's rare that the relationships are completely balanced/equal -- oftentimes one "side" is used much more than another, but you don't really know.  So we could implement both, ie:

`/api/interest-lists/{:id}/experiences`
and
`/api/experiences/{:id}/interest-lists`

...and then we can use either one depending on which makes sense for the use case at-hand.

Bonus for this approach, you can confirm the existence of a relationship by doing

`GET /api/experiences/{:id}/interest-lists/{:id}` which returns 200 OK if present, 404 Not Found if not 

If you were to POST a set of interest list IDs to `/api/experiences/{:id}/interest-lists` as a comma-delimited list of IDs or (probably better) a JSON array of objects it could iterate over them upon receipt.